cli: TestZip failed

[(cli).TestZip failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1729104&tab=buildLog) on [master@c7a19fa36c219936277aca80e2f3c0c89c832c58](https://github.com/cockroachdb/cockroach/commits/c7a19fa36c219936277aca80e2f3c0c89c832c58):

```
W200204 20:32:40.697292 17202417 storage/node_liveness.go:546  [n1,liveness-hb] slow heartbeat took 11.1s
W200204 20:32:40.697346 17202417 storage/node_liveness.go:471  [n1,liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 4.5s
W200204 20:32:41.818995 17202368 storage/store_raft.go:496  [n1,s1,r1/1:/{Min-System/NodeL…}] handle raft ready: 1.1s [applied=1, batches=1, state_assertions=0]
W200204 20:32:41.819281 17202329 storage/store_raft.go:496  [n1,s1,r6/1:/Table/{SystemCon…-11}] handle raft ready: 1.1s [applied=1, batches=1, state_assertions=0]
W200204 20:32:42.396319 17202350 storage/store_raft.go:496  [n1,s1,r2/1:/System/NodeLiveness{-Max}] handle raft ready: 1.7s [applied=1, batches=1, state_assertions=1]
W200204 20:32:42.468701 17202368 storage/store_raft.go:496  [n1,s1,r1/1:/{Min-System/NodeL…}] handle raft ready: 0.6s [applied=1, batches=1, state_assertions=1]
W200204 20:32:44.275590 17202377 storage/store_raft.go:496  [n1,s1,r2/1:/System/NodeLiveness{-Max}] handle raft ready: 1.8s [applied=1, batches=1, state_assertions=0]
W200204 20:32:44.275623 17203962 storage/node_liveness.go:546  [n1,s1,r4/1:/System{/tsd-tse}] slow heartbeat took 6.4s
E200204 20:32:44.275654 17203962 storage/replica_range_lease.go:339  [n1,s1,r4/1:/System{/tsd-tse}] heartbeat failed on epoch increment
W200204 20:32:44.275657 17203942 storage/node_liveness.go:546  [n1,s1,r3/1:/System/{NodeLive…-tsd}] slow heartbeat took 6.4s
E200204 20:32:44.275682 17203942 storage/replica_range_lease.go:339  [n1,s1,r3/1:/System/{NodeLive…-tsd}] heartbeat failed on epoch increment
W200204 20:32:44.275967 17203975 storage/node_liveness.go:546  [n1,s1,r26/1:/NamespaceTable/{30-Max}] slow heartbeat took 5.4s
E200204 20:32:44.275988 17203975 storage/replica_range_lease.go:339  [n1,s1,r26/1:/NamespaceTable/{30-Max}] heartbeat failed on epoch increment
W200204 20:32:44.276084 17203923 storage/node_liveness.go:546  [n1,s1,r6/1:/Table/{SystemCon…-11}] slow heartbeat took 4.2s
E200204 20:32:44.276107 17203923 storage/replica_range_lease.go:339  [n1,s1,r6/1:/Table/{SystemCon…-11}] heartbeat failed on epoch increment
W200204 20:32:44.276181 17204033 storage/node_liveness.go:546  [n1,s1,r10/1:/Table/1{4-5}] slow heartbeat took 3.6s
E200204 20:32:44.276198 17204033 storage/replica_range_lease.go:339  [n1,s1,r10/1:/Table/1{4-5}] heartbeat failed on epoch increment
W200204 20:32:44.276391 17202417 storage/node_liveness.go:546  [n1,liveness-hb] slow heartbeat took 3.6s
W200204 20:32:45.197635 17202417 storage/node_liveness.go:471  [n1,liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 4.5s
W200204 20:32:45.310117 17202357 storage/store_raft.go:496  [n1,s1,r24/1:/Table/2{8-9}] handle raft ready: 1.0s [applied=1, batches=1, state_assertions=1]
W200204 20:32:45.396347 17202373 storage/store_raft.go:496  [n1,s1,r2/1:/System/NodeLiveness{-Max}] handle raft ready: 1.1s [applied=1, batches=1, state_assertions=0]
W200204 20:32:45.396733 17204279 storage/node_liveness.go:546  [n1,s1,r3/1:/System/{NodeLive…-tsd}] slow heartbeat took 1.1s
W200204 20:32:45.397645 17204308 storage/node_liveness.go:546  [n1,s1,r4/1:/System{/tsd-tse}] slow heartbeat took 1.1s
W200204 20:32:45.400436 17204292 storage/node_liveness.go:546  [n1,s1,r26/1:/NamespaceTable/{30-Max}] slow heartbeat took 1.1s
W200204 20:32:51.018495 17202348 storage/store_raft.go:496  [n1,s1,r14/1:/Table/1{8-9}] handle raft ready: 5.6s [applied=1, batches=1, state_assertions=1]
W200204 20:32:51.019644 17202359 storage/store_raft.go:496  [n1,s1,r4/1:/System{/tsd-tse}] handle raft ready: 5.6s [applied=1, batches=1, state_assertions=0]
W200204 20:32:53.272515 17202331 storage/store_raft.go:496  [n1,s1,r2/1:/System/NodeLiveness{-Max}] handle raft ready: 5.7s [applied=1, batches=1, state_assertions=0]
W200204 20:32:53.272774 17202372 storage/store_raft.go:496  [n1,s1,r24/1:/Table/2{8-9}] handle raft ready: 7.8s [applied=1, batches=1, state_assertions=0]
W200204 20:32:53.273588 17202367 storage/store_raft.go:496  [n1,s1,r5/1:/{Systemtse-Table/System…}] handle raft ready: 3.2s [applied=1, batches=1, state_assertions=1]
W200204 20:32:53.274154 17202378 storage/store_raft.go:496  [n1,s1,r3/1:/System/{NodeLive…-tsd}] handle raft ready: 5.4s [applied=1, batches=1, state_assertions=0]
W200204 20:32:53.297494 17202352 storage/store_raft.go:496  [n1,s1,r26/1:/NamespaceTable/{30-Max}] handle raft ready: 0.5s [applied=1, batches=1, state_assertions=1]
W200204 20:32:53.299291 17202359 storage/store_raft.go:496  [n1,s1,r4/1:/System{/tsd-tse}] handle raft ready: 2.3s [applied=2, batches=1, state_assertions=0]
W200204 20:32:53.299390 17202390 storage/store_raft.go:496  [n1,s1,r1/1:/{Min-System/NodeL…}] handle raft ready: 2.3s [applied=1, batches=1, state_assertions=1]
W200204 20:32:53.312668 17202417 storage/node_liveness.go:546  [n1,liveness-hb] slow heartbeat took 5.8s
W200204 20:32:53.312718 17202417 storage/node_liveness.go:471  [n1,liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 4.5s
--- FAIL: TestZip (115.61s)
    zip_test.go:199: 
        	Error Trace:	zip_test.go:199
        	Error:      	Not equal: 
        	            	expected: "debug zip /dev/null\nwriting /dev/null\n  debug/events.json\n  debug/rangelog.json\n  debug/liveness.json\n  debug/settings.json\n  debug/reports/problemranges.json\n  debug/crdb_internal.cluster_queries.txt\n  debug/crdb_internal.cluster_sessions.txt\n  debug/crdb_internal.cluster_settings.txt\n  debug/crdb_internal.jobs.txt\n  debug/system.jobs.txt\n  debug/system.descriptor.txt\n  debug/system.namespace.txt\n  debug/system.namespace_deprecated.txt\n  debug/crdb_internal.kv_node_status.txt\n  debug/crdb_internal.kv_store_status.txt\n  debug/crdb_internal.schema_changes.txt\n  debug/crdb_internal.partitions.txt\n  debug/crdb_internal.zones.txt\n  debug/nodes/1/status.json\n  debug/nodes/1/crdb_internal.feature_usage.txt\n  debug/nodes/1/crdb_internal.gossip_alerts.txt\n  debug/nodes/1/crdb_internal.gossip_liveness.txt\n  debug/nodes/1/crdb_internal.gossip_network.txt\n  debug/nodes/1/crdb_internal.gossip_nodes.txt\n  debug/nodes/1/crdb_internal.leases.txt\n  debug/nodes/1/crdb_internal.node_build_info.txt\n  debug/nodes/1/crdb_internal.node_metrics.txt\n  debug/nodes/1/crdb_internal.node_queries.txt\n  debug/nodes/1/crdb_internal.node_runtime_info.txt\n  debug/nodes/1/crdb_internal.node_sessions.txt\n  debug/nodes/1/crdb_internal.node_statement_statistics.txt\n  debug/nodes/1/crdb_internal.node_txn_stats.txt\n  debug/nodes/1/details.json\n  debug/nodes/1/gossip.json\n  debug/nodes/1/enginestats.json\n  debug/nodes/1/stacks.txt\n  debug/nodes/1/heap.pprof\n  debug/nodes/1/ranges/1.json\n  debug/nodes/1/ranges/2.json\n  debug/nodes/1/ranges/3.json\n  debug/nodes/1/ranges/4.json\n  debug/nodes/1/ranges/5.json\n  debug/nodes/1/ranges/6.json\n  debug/nodes/1/ranges/7.json\n  debug/nodes/1/ranges/8.json\n  debug/nodes/1/ranges/9.json\n  debug/nodes/1/ranges/10.json\n  debug/nodes/1/ranges/11.json\n  debug/nodes/1/ranges/12.json\n  debug/nodes/1/ranges/13.json\n  debug/nodes/1/ranges/14.json\n  debug/nodes/1/ranges/15.json\n  debug/nodes/1/ranges/16.json\n  debug/nodes/1/ranges/17.json\n  debug/nodes/1/ranges/18.json\n  debug/nodes/1/ranges/19.json\n  debug/nodes/1/ranges/20.json\n  debug/nodes/1/ranges/21.json\n  debug/nodes/1/ranges/22.json\n  debug/nodes/1/ranges/23.json\n  debug/nodes/1/ranges/24.json\n  debug/nodes/1/ranges/25.json\n  debug/nodes/1/ranges/26.json\n  debug/nodes/1/ranges/27.json\n  debug/nodes/1/ranges/28.json\n  debug/schema/defaultdb@details.json\n  debug/schema/postgres@details.json\n  debug/schema/system@details.json\n  debug/schema/system/comments.json\n  debug/schema/system/descriptor.json\n  debug/schema/system/eventlog.json\n  debug/schema/system/jobs.json\n  debug/schema/system/lease.json\n  debug/schema/system/locations.json\n  debug/schema/system/namespace.json\n  debug/schema/system/namespace_deprecated.json\n  debug/schema/system/protected_ts_meta.json\n  debug/schema/system/protected_ts_records.json\n  debug/schema/system/rangelog.json\n  debug/schema/system/replication_constraint_stats.json\n  debug/schema/system/replication_critical_localities.json\n  debug/schema/system/replication_stats.json\n  debug/schema/system/reports_meta.json\n  debug/schema/system/role_members.json\n  debug/schema/system/settings.json\n  debug/schema/system/table_statistics.json\n  debug/schema/system/ui.json\n  debug/schema/system/users.json\n  debug/schema/system/web_sessions.json\n  debug/schema/system/zones.json\n"
        	            	actual  : "debug zip /dev/null\nwriting /dev/null\n  debug/events.json\n  debug/rangelog.json\n  debug/liveness.json\n  debug/settings.json\n  debug/reports/problemranges.json\n  debug/crdb_internal.cluster_queries.txt\n  debug/crdb_internal.cluster_sessions.txt\n  debug/crdb_internal.cluster_settings.txt\n  debug/crdb_internal.jobs.txt\n  debug/system.jobs.txt\n  debug/system.descriptor.txt\n  debug/system.namespace.txt\n  debug/system.namespace_deprecated.txt\n  debug/crdb_internal.kv_node_status.txt\n  debug/crdb_internal.kv_store_status.txt\n  debug/crdb_internal.schema_changes.txt\n  debug/crdb_internal.partitions.txt\n  debug/crdb_internal.zones.txt\n  debug/nodes/1/status.json\n  debug/nodes/1/crdb_internal.feature_usage.txt\n  debug/nodes/1/crdb_internal.gossip_alerts.txt\n  debug/nodes/1/crdb_internal.gossip_liveness.txt\n  debug/nodes/1/crdb_internal.gossip_network.txt\n  debug/nodes/1/crdb_internal.gossip_nodes.txt\n  debug/nodes/1/crdb_internal.leases.txt\n  debug/nodes/1/crdb_internal.node_build_info.txt\n  debug/nodes/1/crdb_internal.node_metrics.txt\n  debug/nodes/1/crdb_internal.node_queries.txt\n  debug/nodes/1/crdb_internal.node_runtime_info.txt\n  debug/nodes/1/crdb_internal.node_sessions.txt\n  debug/nodes/1/crdb_internal.node_statement_statistics.txt\n  debug/nodes/1/crdb_internal.node_txn_stats.txt\n  debug/nodes/1/details.json\n  debug/nodes/1/gossip.json\n  debug/nodes/1/enginestats.json\n  debug/nodes/1/stacks.txt\n  debug/nodes/1/heap.pprof\n  debug/nodes/1/ranges/1.json\n  debug/nodes/1/ranges/2.json\n  debug/nodes/1/ranges/3.json\n  debug/nodes/1/ranges/4.json\n  debug/nodes/1/ranges/5.json\n  debug/nodes/1/ranges/6.json\n  debug/nodes/1/ranges/7.json\n  debug/nodes/1/ranges/8.json\n  debug/nodes/1/ranges/9.json\n  debug/nodes/1/ranges/10.json\n  debug/nodes/1/ranges/11.json\n  debug/nodes/1/ranges/12.json\n  debug/nodes/1/ranges/13.json\n  debug/nodes/1/ranges/14.json\n  debug/nodes/1/ranges/15.json\n  debug/nodes/1/ranges/16.json\n  debug/nodes/1/ranges/17.json\n  debug/nodes/1/ranges/18.json\n  debug/nodes/1/ranges/19.json\n  debug/nodes/1/ranges/20.json\n  debug/nodes/1/ranges/21.json\n  debug/nodes/1/ranges/22.json\n  debug/nodes/1/ranges/23.json\n  debug/nodes/1/ranges/24.json\n  debug/nodes/1/ranges/25.json\n  debug/nodes/1/ranges/26.json\n  debug/nodes/1/ranges/27.json\n  debug/nodes/1/ranges/28.json\n  debug/schema/defaultdb@details.json\n  debug/schema/postgres@details.json\n  debug/schema/system@details.json\n  debug/schema/system/comments.json\n  ^- resulted in operation \"request table comments\" timed out after 10s: rpc error: code = DeadlineExceeded desc = context deadline exceeded\n  debug/schema/system/descriptor.json\n  debug/schema/system/eventlog.json\n  debug/schema/system/jobs.json\n  debug/schema/system/lease.json\n  debug/schema/system/locations.json\n  debug/schema/system/namespace.json\n  debug/schema/system/namespace_deprecated.json\n  debug/schema/system/protected_ts_meta.json\n  debug/schema/system/protected_ts_records.json\n  debug/schema/system/rangelog.json\n  debug/schema/system/replication_constraint_stats.json\n  debug/schema/system/replication_critical_localities.json\n  debug/schema/system/replication_stats.json\n  debug/schema/system/reports_meta.json\n  debug/schema/system/role_members.json\n  debug/schema/system/settings.json\n  debug/schema/system/table_statistics.json\n  debug/schema/system/ui.json\n  debug/schema/system/users.json\n  debug/schema/system/web_sessions.json\n  debug/schema/system/zones.json\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -71,2 +71,3 @@
        	            	   debug/schema/system/comments.json
        	            	+  ^- resulted in operation "request table comments" timed out after 10s: rpc error: code = DeadlineExceeded desc = context deadline exceeded
        	            	   debug/schema/system/descriptor.json
        	Test:       	TestZip

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestZip PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestZip.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).TestZip failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1800057&tab=buildLog) on [master@86b53944d16b3018db94955175e1f837485d9f0b](https://github.com/cockroachdb/cockroach/commits/86b53944d16b3018db94955175e1f837485d9f0b):

```
W200310 10:51:43.564107 18283256 kv/kvserver/node_liveness.go:546  [n1,s1,r4/1:/System{/tsd-tse}] slow heartbeat took 18.2s
E200310 10:51:43.564127 18283256 kv/kvserver/replica_range_lease.go:339  [n1,s1,r4/1:/System{/tsd-tse}] heartbeat failed on epoch increment
W200310 10:51:43.564316 18282787 kv/kvserver/node_liveness.go:546  [n1,s1,r3/1:/System/{NodeLive…-tsd}] slow heartbeat took 18.1s
E200310 10:51:43.564335 18282787 kv/kvserver/replica_range_lease.go:339  [n1,s1,r3/1:/System/{NodeLive…-tsd}] heartbeat failed on epoch increment
W200310 10:51:43.564859 18283425 kv/kvserver/node_liveness.go:546  [n1,s1,r6/1:/Table/{SystemCon…-11}] slow heartbeat took 12.5s
E200310 10:51:43.564887 18283425 kv/kvserver/replica_range_lease.go:339  [n1,s1,r6/1:/Table/{SystemCon…-11}] heartbeat failed on epoch increment
W200310 10:51:43.565048 18283404 kv/kvserver/node_liveness.go:546  [n1,s1,r28/1:/Table/3{2-3}] slow heartbeat took 3.2s
E200310 10:51:43.565065 18283404 kv/kvserver/replica_range_lease.go:339  [n1,s1,r28/1:/Table/3{2-3}] heartbeat failed on epoch increment
W200310 10:51:43.565835 18283405 kv/kvserver/node_liveness.go:546  [n1,s1,r5/1:/{Systemtse-Table/System…}] slow heartbeat took 3.2s
E200310 10:51:43.565857 18283405 kv/kvserver/replica_range_lease.go:339  [n1,s1,r5/1:/{Systemtse-Table/System…}] heartbeat failed on epoch increment
W200310 10:51:43.567024 18275371 kv/kvserver/node_liveness.go:546  [n1,liveness-hb] slow heartbeat took 3.2s
W200310 10:51:44.793722 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:44.838793 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:44.948123 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:45.170693 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:45.560173 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:46.249303 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:47.331265 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:48.424828 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:49.509696 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:50.545502 18275332 kv/kvserver/store.go:1541  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200310 10:51:50.833958 18275301 kv/kvserver/store_raft.go:496  [n1,s1,r2/1:/System/NodeLiveness{-Max}] handle raft ready: 7.2s [applied=1, batches=1, state_assertions=0]
W200310 10:51:50.834143 18275296 kv/kvserver/store_raft.go:496  [n1,s1,r4/1:/System{/tsd-tse}] handle raft ready: 7.2s [applied=1, batches=1, state_assertions=0]
W200310 10:51:50.834255 18275364 ts/db.go:194  [n1,ts-poll] error writing time series data: result is ambiguous (server shutdown)
W200310 10:51:50.834282 18275364 ts/db.go:197  [n1,ts-poll] node unavailable; try another peer
W200310 10:51:50.834591 18275371 kv/txn.go:590  [n1,liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
W200310 10:51:50.834861 18275371 kv/kvserver/node_liveness.go:546  [n1,liveness-hb] slow heartbeat took 7.2s
W200310 10:51:50.835109 18275371 kv/kvserver/node_liveness.go:471  [n1,liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 4.5s
W200310 10:51:51.621801 18275299 kv/kvserver/store_raft.go:496  [n1,s1,r3/1:/System/{NodeLive…-tsd}] handle raft ready: 8.0s [applied=1, batches=1, state_assertions=0]
W200310 10:51:51.621975 18275326 kv/kvserver/store.go:1730  [n1,s1,r22/1:/Table/2{6-7}] unable to gossip on capacity change: node unavailable; try another peer
W200310 10:51:51.622113 18275326 kv/kvserver/store_raft.go:496  [n1,s1,r22/1:/Table/2{6-7}] handle raft ready: 8.0s [applied=1, batches=1, state_assertions=1]
W200310 10:51:51.622193 18275231 server/node.go:797  [n1,summaries] error recording status summaries: result is ambiguous (server shutdown)
W200310 10:51:51.622204 18275231 server/node.go:797  [n1,summaries] error recording status summaries: node unavailable; try another peer
--- FAIL: TestZip (83.19s)
    zip_test.go:242: 
        	Error Trace:	zip_test.go:242
        	Error:      	Not equal: 
        	            	expected: "debug zip /dev/null\nestablishing RPC connection to ...\nretrieving the node status to get the SQL address...\nusing SQL address: ...\nusing SQL connection URL: postgresql://...\nwriting /dev/null\nrequesting data for debug/events... writing: debug/events.json\nrequesting data for debug/rangelog... writing: debug/rangelog.json\nrequesting data for debug/liveness... writing: debug/liveness.json\nrequesting data for debug/settings... writing: debug/settings.json\nrequesting data for debug/reports/problemranges... writing: debug/reports/problemranges.json\nretrieving SQL data for crdb_internal.cluster_queries... writing: debug/crdb_internal.cluster_queries.txt\nretrieving SQL data for crdb_internal.cluster_sessions... writing: debug/crdb_internal.cluster_sessions.txt\nretrieving SQL data for crdb_internal.cluster_settings... writing: debug/crdb_internal.cluster_settings.txt\nretrieving SQL data for crdb_internal.jobs... writing: debug/crdb_internal.jobs.txt\nretrieving SQL data for system.jobs... writing: debug/system.jobs.txt\nretrieving SQL data for system.descriptor... writing: debug/system.descriptor.txt\nretrieving SQL data for system.namespace... writing: debug/system.namespace.txt\nretrieving SQL data for system.namespace_deprecated... writing: debug/system.namespace_deprecated.txt\nretrieving SQL data for crdb_internal.kv_node_status... writing: debug/crdb_internal.kv_node_status.txt\nretrieving SQL data for crdb_internal.kv_store_status... writing: debug/crdb_internal.kv_store_status.txt\nretrieving SQL data for crdb_internal.schema_changes... writing: debug/crdb_internal.schema_changes.txt\nretrieving SQL data for crdb_internal.partitions... writing: debug/crdb_internal.partitions.txt\nretrieving SQL data for crdb_internal.zones... writing: debug/crdb_internal.zones.txt\nrequesting nodes... writing: debug/nodes/1/status.json\nusing SQL connection URL for node 1: postgresql://...\nretrieving SQL data for crdb_internal.feature_usage... writing: debug/nodes/1/crdb_internal.feature_usage.txt\nretrieving SQL data for crdb_internal.gossip_alerts... writing: debug/nodes/1/crdb_internal.gossip_alerts.txt\nretrieving SQL data for crdb_internal.gossip_liveness... writing: debug/nodes/1/crdb_internal.gossip_liveness.txt\nretrieving SQL data for crdb_internal.gossip_network... writing: debug/nodes/1/crdb_internal.gossip_network.txt\nretrieving SQL data for crdb_internal.gossip_nodes... writing: debug/nodes/1/crdb_internal.gossip_nodes.txt\nretrieving SQL data for crdb_internal.leases... writing: debug/nodes/1/crdb_internal.leases.txt\nretrieving SQL data for crdb_internal.node_build_info... writing: debug/nodes/1/crdb_internal.node_build_info.txt\nretrieving SQL data for crdb_internal.node_metrics... writing: debug/nodes/1/crdb_internal.node_metrics.txt\nretrieving SQL data for crdb_internal.node_queries... writing: debug/nodes/1/crdb_internal.node_queries.txt\nretrieving SQL data for crdb_internal.node_runtime_info... writing: debug/nodes/1/crdb_internal.node_runtime_info.txt\nretrieving SQL data for crdb_internal.node_sessions... writing: debug/nodes/1/crdb_internal.node_sessions.txt\nretrieving SQL data for crdb_internal.node_statement_statistics... writing: debug/nodes/1/crdb_internal.node_statement_statistics.txt\nretrieving SQL data for crdb_internal.node_txn_stats... writing: debug/nodes/1/crdb_internal.node_txn_stats.txt\nrequesting data for debug/nodes/1/details... writing: debug/nodes/1/details.json\nrequesting data for debug/nodes/1/gossip... writing: debug/nodes/1/gossip.json\nrequesting data for debug/nodes/1/enginestats... writing: debug/nodes/1/enginestats.json\nrequesting stacks for node 1... writing: debug/nodes/1/stacks.txt\nrequesting threads for node 1... writing: debug/nodes/1/threads.txt\nrequesting heap profile for node 1... writing: debug/nodes/1/heap.pprof\nrequesting heap files for node 1... 0 found\nrequesting goroutine files for node 1... 0 found\nrequesting log file ...\nrequesting ranges... 32 found\nwriting: debug/nodes/1/ranges/1.json\nwriting: debug/nodes/1/ranges/2.json\nwriting: debug/nodes/1/ranges/3.json\nwriting: debug/nodes/1/ranges/4.json\nwriting: debug/nodes/1/ranges/5.json\nwriting: debug/nodes/1/ranges/6.json\nwriting: debug/nodes/1/ranges/7.json\nwriting: debug/nodes/1/ranges/8.json\nwriting: debug/nodes/1/ranges/9.json\nwriting: debug/nodes/1/ranges/10.json\nwriting: debug/nodes/1/ranges/11.json\nwriting: debug/nodes/1/ranges/12.json\nwriting: debug/nodes/1/ranges/13.json\nwriting: debug/nodes/1/ranges/14.json\nwriting: debug/nodes/1/ranges/15.json\nwriting: debug/nodes/1/ranges/16.json\nwriting: debug/nodes/1/ranges/17.json\nwriting: debug/nodes/1/ranges/18.json\nwriting: debug/nodes/1/ranges/19.json\nwriting: debug/nodes/1/ranges/20.json\nwriting: debug/nodes/1/ranges/21.json\nwriting: debug/nodes/1/ranges/22.json\nwriting: debug/nodes/1/ranges/23.json\nwriting: debug/nodes/1/ranges/24.json\nwriting: debug/nodes/1/ranges/25.json\nwriting: debug/nodes/1/ranges/26.json\nwriting: debug/nodes/1/ranges/27.json\nwriting: debug/nodes/1/ranges/28.json\nwriting: debug/nodes/1/ranges/29.json\nwriting: debug/nodes/1/ranges/30.json\nwriting: debug/nodes/1/ranges/31.json\nwriting: debug/nodes/1/ranges/32.json\nrequesting list of SQL databases... 3 found\nrequesting database details for defaultdb... writing: debug/schema/defaultdb@details.json\n0 tables found\nrequesting database details for postgres... writing: debug/schema/postgres@details.json\n0 tables found\nrequesting database details for system... writing: debug/schema/system@details.json\n26 tables found\nrequesting table details for system.comments... writing: debug/schema/system/comments.json\nrequesting table details for system.descriptor... writing: debug/schema/system/descriptor.json\nrequesting table details for system.eventlog... writing: debug/schema/system/eventlog.json\nrequesting table details for system.jobs... writing: debug/schema/system/jobs.json\nrequesting table details for system.lease... writing: debug/schema/system/lease.json\nrequesting table details for system.locations... writing: debug/schema/system/locations.json\nrequesting table details for system.namespace... writing: debug/schema/system/namespace.json\nrequesting table details for system.namespace_deprecated... writing: debug/schema/system/namespace_deprecated.json\nrequesting table details for system.protected_ts_meta... writing: debug/schema/system/protected_ts_meta.json\nrequesting table details for system.protected_ts_records... writing: debug/schema/system/protected_ts_records.json\nrequesting table details for system.rangelog... writing: debug/schema/system/rangelog.json\nrequesting table details for system.replication_constraint_stats... writing: debug/schema/system/replication_constraint_stats.json\nrequesting table details for system.replication_critical_localities... writing: debug/schema/system/replication_critical_localities.json\nrequesting table details for system.replication_stats... writing: debug/schema/system/replication_stats.json\nrequesting table details for system.reports_meta... writing: debug/schema/system/reports_meta.json\nrequesting table details for system.role_members... writing: debug/schema/system/role_members.json\nrequesting table details for system.role_options... writing: debug/schema/system/role_options.json\nrequesting table details for system.settings... writing: debug/schema/system/settings.json\nrequesting table details for system.statement_bundle_chunks... writing: debug/schema/system/statement_bundle_chunks.json\nrequesting table details for system.statement_diagnostics... writing: debug/schema/system/statement_diagnostics.json\nrequesting table details for system.statement_diagnostics_requests... writing: debug/schema/system/statement_diagnostics_requests.json\nrequesting table details for system.table_statistics... writing: debug/schema/system/table_statistics.json\nrequesting table details for system.ui... writing: debug/schema/system/ui.json\nrequesting table details for system.users... writing: debug/schema/system/users.json\nrequesting table details for system.web_sessions... writing: debug/schema/system/web_sessions.json\nrequesting table details for system.zones... writing: debug/schema/system/zones.json\n"
        	            	actual  : "debug zip /dev/null\nestablishing RPC connection to ...\nretrieving the node status to get the SQL address...\nusing SQL address: ...\nusing SQL connection URL: postgresql://...\nwriting /dev/null\nrequesting data for debug/events... writing: debug/events.json\nrequesting data for debug/rangelog... writing: debug/rangelog.json\nrequesting data for debug/liveness... writing: debug/liveness.json\nrequesting data for debug/settings... writing: debug/settings.json\nrequesting data for debug/reports/problemranges... writing: debug/reports/problemranges.json\nretrieving SQL data for crdb_internal.cluster_queries... writing: debug/crdb_internal.cluster_queries.txt\nretrieving SQL data for crdb_internal.cluster_sessions... writing: debug/crdb_internal.cluster_sessions.txt\nretrieving SQL data for crdb_internal.cluster_settings... writing: debug/crdb_internal.cluster_settings.txt\nretrieving SQL data for crdb_internal.jobs... writing: debug/crdb_internal.jobs.txt\nretrieving SQL data for system.jobs... writing: debug/system.jobs.txt\nretrieving SQL data for system.descriptor... writing: debug/system.descriptor.txt\nretrieving SQL data for system.namespace... writing: debug/system.namespace.txt\nretrieving SQL data for system.namespace_deprecated... writing: debug/system.namespace_deprecated.txt\nretrieving SQL data for crdb_internal.kv_node_status... writing: debug/crdb_internal.kv_node_status.txt\nretrieving SQL data for crdb_internal.kv_store_status... writing: debug/crdb_internal.kv_store_status.txt\nretrieving SQL data for crdb_internal.schema_changes... writing: debug/crdb_internal.schema_changes.txt\nretrieving SQL data for crdb_internal.partitions... writing: debug/crdb_internal.partitions.txt\nretrieving SQL data for crdb_internal.zones... writing: debug/crdb_internal.zones.txt\nrequesting nodes... writing: debug/nodes/1/status.json\nusing SQL connection URL for node 1: postgresql://...\nretrieving SQL data for crdb_internal.feature_usage... writing: debug/nodes/1/crdb_internal.feature_usage.txt\nretrieving SQL data for crdb_internal.gossip_alerts... writing: debug/nodes/1/crdb_internal.gossip_alerts.txt\nretrieving SQL data for crdb_internal.gossip_liveness... writing: debug/nodes/1/crdb_internal.gossip_liveness.txt\nretrieving SQL data for crdb_internal.gossip_network... writing: debug/nodes/1/crdb_internal.gossip_network.txt\nretrieving SQL data for crdb_internal.gossip_nodes... writing: debug/nodes/1/crdb_internal.gossip_nodes.txt\nretrieving SQL data for crdb_internal.leases... writing: debug/nodes/1/crdb_internal.leases.txt\nretrieving SQL data for crdb_internal.node_build_info... writing: debug/nodes/1/crdb_internal.node_build_info.txt\nretrieving SQL data for crdb_internal.node_metrics... writing: debug/nodes/1/crdb_internal.node_metrics.txt\nretrieving SQL data for crdb_internal.node_queries... writing: debug/nodes/1/crdb_internal.node_queries.txt\nretrieving SQL data for crdb_internal.node_runtime_info... writing: debug/nodes/1/crdb_internal.node_runtime_info.txt\nretrieving SQL data for crdb_internal.node_sessions... writing: debug/nodes/1/crdb_internal.node_sessions.txt\nretrieving SQL data for crdb_internal.node_statement_statistics... writing: debug/nodes/1/crdb_internal.node_statement_statistics.txt\nretrieving SQL data for crdb_internal.node_txn_stats... writing: debug/nodes/1/crdb_internal.node_txn_stats.txt\nrequesting data for debug/nodes/1/details... writing: debug/nodes/1/details.json\nrequesting data for debug/nodes/1/gossip... writing: debug/nodes/1/gossip.json\nrequesting data for debug/nodes/1/enginestats... writing: debug/nodes/1/enginestats.json\nrequesting stacks for node 1... writing: debug/nodes/1/stacks.txt\nrequesting threads for node 1... writing: debug/nodes/1/threads.txt\nrequesting heap profile for node 1... writing: debug/nodes/1/heap.pprof\nrequesting heap files for node 1... 0 found\nrequesting goroutine files for node 1... 0 found\nrequesting log file ...\nrequesting ranges... 32 found\nwriting: debug/nodes/1/ranges/1.json\nwriting: debug/nodes/1/ranges/2.json\nwriting: debug/nodes/1/ranges/3.json\nwriting: debug/nodes/1/ranges/4.json\nwriting: debug/nodes/1/ranges/5.json\nwriting: debug/nodes/1/ranges/6.json\nwriting: debug/nodes/1/ranges/7.json\nwriting: debug/nodes/1/ranges/8.json\nwriting: debug/nodes/1/ranges/9.json\nwriting: debug/nodes/1/ranges/10.json\nwriting: debug/nodes/1/ranges/11.json\nwriting: debug/nodes/1/ranges/12.json\nwriting: debug/nodes/1/ranges/13.json\nwriting: debug/nodes/1/ranges/14.json\nwriting: debug/nodes/1/ranges/15.json\nwriting: debug/nodes/1/ranges/16.json\nwriting: debug/nodes/1/ranges/17.json\nwriting: debug/nodes/1/ranges/18.json\nwriting: debug/nodes/1/ranges/19.json\nwriting: debug/nodes/1/ranges/20.json\nwriting: debug/nodes/1/ranges/21.json\nwriting: debug/nodes/1/ranges/22.json\nwriting: debug/nodes/1/ranges/23.json\nwriting: debug/nodes/1/ranges/24.json\nwriting: debug/nodes/1/ranges/25.json\nwriting: debug/nodes/1/ranges/26.json\nwriting: debug/nodes/1/ranges/27.json\nwriting: debug/nodes/1/ranges/28.json\nwriting: debug/nodes/1/ranges/29.json\nwriting: debug/nodes/1/ranges/30.json\nwriting: debug/nodes/1/ranges/31.json\nwriting: debug/nodes/1/ranges/32.json\nrequesting list of SQL databases... 3 found\nrequesting database details for defaultdb... writing: debug/schema/defaultdb@details.json\n0 tables found\nrequesting database details for postgres... writing: debug/schema/postgres@details.json\n0 tables found\nrequesting database details for system... writing: debug/schema/system@details.json\n26 tables found\nrequesting table details for system.comments... writing: debug/schema/system/comments.json\nrequesting table details for system.descriptor... writing: debug/schema/system/descriptor.json\nrequesting table details for system.eventlog... writing: debug/schema/system/eventlog.json\nrequesting table details for system.jobs... writing: debug/schema/system/jobs.json\nrequesting table details for system.lease... writing: debug/schema/system/lease.json\nrequesting table details for system.locations... writing: debug/schema/system/locations.json\nrequesting table details for system.namespace... writing: debug/schema/system/namespace.json\nrequesting table details for system.namespace_deprecated... writing: debug/schema/system/namespace_deprecated.json\nrequesting table details for system.protected_ts_meta... writing: debug/schema/system/protected_ts_meta.json\nrequesting table details for system.protected_ts_records... writing: debug/schema/system/protected_ts_records.json\nrequesting table details for system.rangelog... writing: debug/schema/system/rangelog.json\nrequesting table details for system.replication_constraint_stats... writing: debug/schema/system/replication_constraint_stats.json\nrequesting table details for system.replication_critical_localities... writing: debug/schema/system/replication_critical_localities.json\nrequesting table details for system.replication_stats... writing: debug/schema/system/replication_stats.json\nrequesting table details for system.reports_meta... writing: debug/schema/system/reports_meta.json\nrequesting table details for system.role_members... writing: debug/schema/system/role_members.json\nrequesting table details for system.role_options... writing: debug/schema/system/role_options.json\nrequesting table details for system.settings... writing: debug/schema/system/settings.json\nrequesting table details for system.statement_bundle_chunks... writing: debug/schema/system/statement_bundle_chunks.json\nrequesting table details for system.statement_diagnostics... writing: debug/schema/system/statement_diagnostics.json\nrequesting table details for system.statement_diagnostics_requests... writing: debug/schema/system/statement_diagnostics_requests.json\nrequesting table details for system.table_statistics... writing: debug/schema/system/table_statistics.json\nrequesting table details for system.ui... writing: debug/schema/system/ui.json\n  ^- resulted in operation \"requesting table details for system.ui\" timed out after 10s: rpc error: code = DeadlineExceeded desc = context deadline exceeded\nrequesting table details for system.users... writing: debug/schema/system/users.json\n  ^- resulted in operation \"requesting table details for system.users\" timed out after 10s: rpc error: code = DeadlineExceeded desc = context deadline exceeded\nrequesting table details for system.web_sessions... writing: debug/schema/system/web_sessions.json\nrequesting table details for system.zones... writing: debug/schema/system/zones.json\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -111,3 +111,5 @@
        	            	 requesting table details for system.ui... writing: debug/schema/system/ui.json
        	            	+  ^- resulted in operation "requesting table details for system.ui" timed out after 10s: rpc error: code = DeadlineExceeded desc = context deadline exceeded
        	            	 requesting table details for system.users... writing: debug/schema/system/users.json
        	            	+  ^- resulted in operation "requesting table details for system.users" timed out after 10s: rpc error: code = DeadlineExceeded desc = context deadline exceeded
        	            	 requesting table details for system.web_sessions... writing: debug/schema/system/web_sessions.json
        	Test:       	TestZip

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestZip PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestZip.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(cli).TestZip failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1891076&tab=buildLog) on [master@0714b8693ccfafdf25c8ca58f435b4ffae096623](https://github.com/cockroachdb/cockroach/commits/0714b8693ccfafdf25c8ca58f435b4ffae096623):

```
	/usr/local/go/src/runtime/proc.go:190
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1357
E200422 02:37:41.920148 19449816 server/admin.go:351  admin-query-namespace-ID: context deadline exceeded
W200422 02:37:44.658758 19448069 kv/kvserver/node_liveness.go:571  [n1,liveness-hb] slow heartbeat took 9.0s
W200422 02:37:44.658843 19448069 kv/kvserver/node_liveness.go:496  [n1,liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 4.5s
(1) operation "node liveness heartbeat" timed out after 4.5s
Wraps: (2) context deadline exceeded
Error types: (1) *contextutil.TimeoutError (2) context.deadlineExceededError
W200422 02:37:44.659112 19448008 kv/kvserver/store_raft.go:508  [n1,s1,r3/1:/System/{NodeLive…-tsd}] handle raft ready: 6.8s [applied=1, batches=1, state_assertions=0]
W200422 02:37:44.659212 19448016 kv/kvserver/store_raft.go:508  [n1,s1,r23/1:/Table/2{7-8}] handle raft ready: 9.1s [applied=5, batches=1, state_assertions=0]
W200422 02:37:44.661271 19448025 kv/kvserver/store_raft.go:508  [n1,s1,r4/1:/System{/tsd-tse}] handle raft ready: 6.8s [applied=1, batches=1, state_assertions=0]
W200422 02:37:44.727193 19448003 kv/kvserver/store_raft.go:508  [n1,s1,r7/1:/Table/1{1-2}] handle raft ready: 5.5s [applied=2, batches=1, state_assertions=0]
W200422 02:37:44.728098 19448029 kv/kvserver/store_raft.go:508  [n1,s1,r24/1:/Table/2{8-9}] handle raft ready: 5.5s [applied=1, batches=1, state_assertions=0]
W200422 02:37:44.728677 19448026 kv/kvserver/store_raft.go:508  [n1,s1,r1/1:/{Min-System/NodeL…}] handle raft ready: 5.5s [applied=1, batches=1, state_assertions=0]
W200422 02:37:44.779932 19449869 kv/kvserver/node_liveness.go:571  [n1,s1,r26/1:/NamespaceTable/{30-Max}] slow heartbeat took 2.9s
W200422 02:37:53.639821 19448053 kv/kvserver/store.go:1627  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200422 02:37:53.691025 19448053 kv/kvserver/store.go:1627  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200422 02:37:53.791506 19448053 kv/kvserver/store.go:1627  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200422 02:37:53.967730 19448053 kv/kvserver/store.go:1627  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200422 02:37:54.310923 19448053 kv/kvserver/store.go:1627  [n1,s1,r1/1:/{Min-System/NodeL…}] could not gossip first range descriptor: node unavailable; try another peer
W200422 02:37:54.804959 19448026 kv/kvserver/store_raft.go:508  [n1,s1,r1/1:/{Min-System/NodeL…}] handle raft ready: 3.4s [applied=1, batches=1, state_assertions=1]
W200422 02:38:01.359681 19448026 kv/kvserver/store_raft.go:508  [n1,s1,r1/1:/{Min-System/NodeL…}] handle raft ready: 6.6s [applied=1, batches=1, state_assertions=0]
--- FAIL: TestZip (108.04s)
    zip_test.go:246: 
        	Error Trace:	zip_test.go:246
        	Error:      	Not equal: 
        	            	expected: "debug zip /dev/null\nestablishing RPC connection to ...\nretrieving the node status to get the SQL address...\nusing SQL address: ...\nusing SQL connection URL: postgresql://...\nwriting /dev/null\nrequesting data for debug/events... writing: debug/events.json\nrequesting data for debug/rangelog... writing: debug/rangelog.json\nrequesting data for debug/liveness... writing: debug/liveness.json\nrequesting data for debug/settings... writing: debug/settings.json\nrequesting data for debug/reports/problemranges... writing: debug/reports/problemranges.json\nretrieving SQL data for crdb_internal.cluster_queries... writing: debug/crdb_internal.cluster_queries.txt\nretrieving SQL data for crdb_internal.cluster_sessions... writing: debug/crdb_internal.cluster_sessions.txt\nretrieving SQL data for crdb_internal.cluster_settings... writing: debug/crdb_internal.cluster_settings.txt\nretrieving SQL data for crdb_internal.cluster_transactions... writing: debug/crdb_internal.cluster_transactions.txt\nretrieving SQL data for crdb_internal.jobs... writing: debug/crdb_internal.jobs.txt\nretrieving SQL data for system.jobs... writing: debug/system.jobs.txt\nretrieving SQL data for system.descriptor... writing: debug/system.descriptor.txt\nretrieving SQL data for system.namespace... writing: debug/system.namespace.txt\nretrieving SQL data for system.namespace2... writing: debug/system.namespace2.txt\nretrieving SQL data for crdb_internal.kv_node_status... writing: debug/crdb_internal.kv_node_status.txt\nretrieving SQL data for crdb_internal.kv_store_status... writing: debug/crdb_internal.kv_store_status.txt\nretrieving SQL data for crdb_internal.schema_changes... writing: debug/crdb_internal.schema_changes.txt\nretrieving SQL data for crdb_internal.partitions... writing: debug/crdb_internal.partitions.txt\nretrieving SQL data for crdb_internal.zones... writing: debug/crdb_internal.zones.txt\nrequesting nodes... writing: debug/nodes/1/status.json\nusing SQL connection URL for node 1: postgresql://...\nretrieving SQL data for crdb_internal.feature_usage... writing: debug/nodes/1/crdb_internal.feature_usage.txt\nretrieving SQL data for crdb_internal.gossip_alerts... writing: debug/nodes/1/crdb_internal.gossip_alerts.txt\nretrieving SQL data for crdb_internal.gossip_liveness... writing: debug/nodes/1/crdb_internal.gossip_liveness.txt\nretrieving SQL data for crdb_internal.gossip_network... writing: debug/nodes/1/crdb_internal.gossip_network.txt\nretrieving SQL data for crdb_internal.gossip_nodes... writing: debug/nodes/1/crdb_internal.gossip_nodes.txt\nretrieving SQL data for crdb_internal.leases... writing: debug/nodes/1/crdb_internal.leases.txt\nretrieving SQL data for crdb_internal.node_build_info... writing: debug/nodes/1/crdb_internal.node_build_info.txt\nretrieving SQL data for crdb_internal.node_metrics... writing: debug/nodes/1/crdb_internal.node_metrics.txt\nretrieving SQL data for crdb_internal.node_queries... writing: debug/nodes/1/crdb_internal.node_queries.txt\nretrieving SQL data for crdb_internal.node_runtime_info... writing: debug/nodes/1/crdb_internal.node_runtime_info.txt\nretrieving SQL data for crdb_internal.node_sessions... writing: debug/nodes/1/crdb_internal.node_sessions.txt\nretrieving SQL data for crdb_internal.node_statement_statistics... writing: debug/nodes/1/crdb_internal.node_statement_statistics.txt\nretrieving SQL data for crdb_internal.node_transactions... writing: debug/nodes/1/crdb_internal.node_transactions.txt\nretrieving SQL data for crdb_internal.node_txn_stats... writing: debug/nodes/1/crdb_internal.node_txn_stats.txt\nrequesting data for debug/nodes/1/details... writing: debug/nodes/1/details.json\nrequesting data for debug/nodes/1/gossip... writing: debug/nodes/1/gossip.json\nrequesting data for debug/nodes/1/enginestats... writing: debug/nodes/1/enginestats.json\nrequesting stacks for node 1... writing: debug/nodes/1/stacks.txt\nrequesting threads for node 1... writing: debug/nodes/1/threads.txt\nrequesting heap profile for node 1... writing: debug/nodes/1/heap.pprof\nrequesting heap files for node 1... 0 found\nrequesting goroutine files for node 1... 0 found\nrequesting log file ...\nrequesting ranges... 32 found\nwriting: debug/nodes/1/ranges/1.json\nwriting: debug/nodes/1/ranges/2.json\nwriting: debug/nodes/1/ranges/3.json\nwriting: debug/nodes/1/ranges/4.json\nwriting: debug/nodes/1/ranges/5.json\nwriting: debug/nodes/1/ranges/6.json\nwriting: debug/nodes/1/ranges/7.json\nwriting: debug/nodes/1/ranges/8.json\nwriting: debug/nodes/1/ranges/9.json\nwriting: debug/nodes/1/ranges/10.json\nwriting: debug/nodes/1/ranges/11.json\nwriting: debug/nodes/1/ranges/12.json\nwriting: debug/nodes/1/ranges/13.json\nwriting: debug/nodes/1/ranges/14.json\nwriting: debug/nodes/1/ranges/15.json\nwriting: debug/nodes/1/ranges/16.json\nwriting: debug/nodes/1/ranges/17.json\nwriting: debug/nodes/1/ranges/18.json\nwriting: debug/nodes/1/ranges/19.json\nwriting: debug/nodes/1/ranges/20.json\nwriting: debug/nodes/1/ranges/21.json\nwriting: debug/nodes/1/ranges/22.json\nwriting: debug/nodes/1/ranges/23.json\nwriting: debug/nodes/1/ranges/24.json\nwriting: debug/nodes/1/ranges/25.json\nwriting: debug/nodes/1/ranges/26.json\nwriting: debug/nodes/1/ranges/27.json\nwriting: debug/nodes/1/ranges/28.json\nwriting: debug/nodes/1/ranges/29.json\nwriting: debug/nodes/1/ranges/30.json\nwriting: debug/nodes/1/ranges/31.json\nwriting: debug/nodes/1/ranges/32.json\nrequesting list of SQL databases... 3 found\nrequesting database details for defaultdb... writing: debug/schema/defaultdb@details.json\n0 tables found\nrequesting database details for postgres... writing: debug/schema/postgres@details.json\n0 tables found\nrequesting database details for system... writing: debug/schema/system@details.json\n26 tables found\nrequesting table details for system.comments... writing: debug/schema/system/comments.json\nrequesting table details for system.descriptor... writing: debug/schema/system/descriptor.json\nrequesting table details for system.eventlog... writing: debug/schema/system/eventlog.json\nrequesting table details for system.jobs... writing: debug/schema/system/jobs.json\nrequesting table details for system.lease... writing: debug/schema/system/lease.json\nrequesting table details for system.locations... writing: debug/schema/system/locations.json\nrequesting table details for system.namespace... writing: debug/schema/system/namespace.json\nrequesting table details for system.namespace2... writing: debug/schema/system/namespace2.json\nrequesting table details for system.protected_ts_meta... writing: debug/schema/system/protected_ts_meta.json\nrequesting table details for system.protected_ts_records... writing: debug/schema/system/protected_ts_records.json\nrequesting table details for system.rangelog... writing: debug/schema/system/rangelog.json\nrequesting table details for system.replication_constraint_stats... writing: debug/schema/system/replication_constraint_stats.json\nrequesting table details for system.replication_critical_localities... writing: debug/schema/system/replication_critical_localities.json\nrequesting table details for system.replication_stats... writing: debug/schema/system/replication_stats.json\nrequesting table details for system.reports_meta... writing: debug/schema/system/reports_meta.json\nrequesting table details for system.role_members... writing: debug/schema/system/role_members.json\nrequesting table details for system.role_options... writing: debug/schema/system/role_options.json\nrequesting table details for system.settings... writing: debug/schema/system/settings.json\nrequesting table details for system.statement_bundle_chunks... writing: debug/schema/system/statement_bundle_chunks.json\nrequesting table details for system.statement_diagnostics... writing: debug/schema/system/statement_diagnostics.json\nrequesting table details for system.statement_diagnostics_requests... writing: debug/schema/system/statement_diagnostics_requests.json\nrequesting table details for system.table_statistics... writing: debug/schema/system/table_statistics.json\nrequesting table details for system.ui... writing: debug/schema/system/ui.json\nrequesting table details for system.users... writing: debug/schema/system/users.json\nrequesting table details for system.web_sessions... writing: debug/schema/system/web_sessions.json\nrequesting table details for system.zones... writing: debug/schema/system/zones.json\n"
        	            	actual  : "debug zip /dev/null\nestablishing RPC connection to ...\nretrieving the node status to get the SQL address...\nusing SQL address: ...\nusing SQL connection URL: postgresql://...\nwriting /dev/null\nrequesting data for debug/events... writing: debug/events.json\nrequesting data for debug/rangelog... writing: debug/rangelog.json\nrequesting data for debug/liveness... writing: debug/liveness.json\nrequesting data for debug/settings... writing: debug/settings.json\nrequesting data for debug/reports/problemranges... writing: debug/reports/problemranges.json\nretrieving SQL data for crdb_internal.cluster_queries... writing: debug/crdb_internal.cluster_queries.txt\nretrieving SQL data for crdb_internal.cluster_sessions... writing: debug/crdb_internal.cluster_sessions.txt\nretrieving SQL data for crdb_internal.cluster_settings... writing: debug/crdb_internal.cluster_settings.txt\nretrieving SQL data for crdb_internal.cluster_transactions... writing: debug/crdb_internal.cluster_transactions.txt\nretrieving SQL data for crdb_internal.jobs... writing: debug/crdb_internal.jobs.txt\nretrieving SQL data for system.jobs... writing: debug/system.jobs.txt\nretrieving SQL data for system.descriptor... writing: debug/system.descriptor.txt\nretrieving SQL data for system.namespace... writing: debug/system.namespace.txt\nretrieving SQL data for system.namespace2... writing: debug/system.namespace2.txt\nretrieving SQL data for crdb_internal.kv_node_status... writing: debug/crdb_internal.kv_node_status.txt\nretrieving SQL data for crdb_internal.kv_store_status... writing: debug/crdb_internal.kv_store_status.txt\nretrieving SQL data for crdb_internal.schema_changes... writing: debug/crdb_internal.schema_changes.txt\nretrieving SQL data for crdb_internal.partitions... writing: debug/crdb_internal.partitions.txt\nretrieving SQL data for crdb_internal.zones... writing: debug/crdb_internal.zones.txt\nrequesting nodes... writing: debug/nodes/1/status.json\nusing SQL connection URL for node 1: postgresql://...\nretrieving SQL data for crdb_internal.feature_usage... writing: debug/nodes/1/crdb_internal.feature_usage.txt\nretrieving SQL data for crdb_internal.gossip_alerts... writing: debug/nodes/1/crdb_internal.gossip_alerts.txt\nretrieving SQL data for crdb_internal.gossip_liveness... writing: debug/nodes/1/crdb_internal.gossip_liveness.txt\nretrieving SQL data for crdb_internal.gossip_network... writing: debug/nodes/1/crdb_internal.gossip_network.txt\nretrieving SQL data for crdb_internal.gossip_nodes... writing: debug/nodes/1/crdb_internal.gossip_nodes.txt\nretrieving SQL data for crdb_internal.leases... writing: debug/nodes/1/crdb_internal.leases.txt\nretrieving SQL data for crdb_internal.node_build_info... writing: debug/nodes/1/crdb_internal.node_build_info.txt\nretrieving SQL data for crdb_internal.node_metrics... writing: debug/nodes/1/crdb_internal.node_metrics.txt\nretrieving SQL data for crdb_internal.node_queries... writing: debug/nodes/1/crdb_internal.node_queries.txt\nretrieving SQL data for crdb_internal.node_runtime_info... writing: debug/nodes/1/crdb_internal.node_runtime_info.txt\nretrieving SQL data for crdb_internal.node_sessions... writing: debug/nodes/1/crdb_internal.node_sessions.txt\nretrieving SQL data for crdb_internal.node_statement_statistics... writing: debug/nodes/1/crdb_internal.node_statement_statistics.txt\nretrieving SQL data for crdb_internal.node_transactions... writing: debug/nodes/1/crdb_internal.node_transactions.txt\nretrieving SQL data for crdb_internal.node_txn_stats... writing: debug/nodes/1/crdb_internal.node_txn_stats.txt\nrequesting data for debug/nodes/1/details... writing: debug/nodes/1/details.json\nrequesting data for debug/nodes/1/gossip... writing: debug/nodes/1/gossip.json\nrequesting data for debug/nodes/1/enginestats... writing: debug/nodes/1/enginestats.json\nrequesting stacks for node 1... writing: debug/nodes/1/stacks.txt\nrequesting threads for node 1... writing: debug/nodes/1/threads.txt\nrequesting heap profile for node 1... writing: debug/nodes/1/heap.pprof\nrequesting heap files for node 1... 4 found\nwriting: debug/nodes/1/heapprof/memprof.000000000039162280_2020-04-22T02_36_38.141.pprof\nwriting: debug/nodes/1/heapprof/memprof.000000000055517712_2020-04-22T02_36_48.142.pprof\nwriting: debug/nodes/1/heapprof/memprof.000000000064787432_2020-04-22T02_36_58.142.pprof\nwriting: debug/nodes/1/heapprof/memprof.000000000065956448_2020-04-22T02_37_18.142.pprof\nrequesting goroutine files for node 1... 0 found\nrequesting log file ...\nrequesting ranges... 32 found\nwriting: debug/nodes/1/ranges/1.json\nwriting: debug/nodes/1/ranges/2.json\nwriting: debug/nodes/1/ranges/3.json\nwriting: debug/nodes/1/ranges/4.json\nwriting: debug/nodes/1/ranges/5.json\nwriting: debug/nodes/1/ranges/6.json\nwriting: debug/nodes/1/ranges/7.json\nwriting: debug/nodes/1/ranges/8.json\nwriting: debug/nodes/1/ranges/9.json\nwriting: debug/nodes/1/ranges/10.json\nwriting: debug/nodes/1/ranges/11.json\nwriting: debug/nodes/1/ranges/12.json\nwriting: debug/nodes/1/ranges/13.json\nwriting: debug/nodes/1/ranges/14.json\nwriting: debug/nodes/1/ranges/15.json\nwriting: debug/nodes/1/ranges/16.json\nwriting: debug/nodes/1/ranges/17.json\nwriting: debug/nodes/1/ranges/18.json\nwriting: debug/nodes/1/ranges/19.json\nwriting: debug/nodes/1/ranges/20.json\nwriting: debug/nodes/1/ranges/21.json\nwriting: debug/nodes/1/ranges/22.json\nwriting: debug/nodes/1/ranges/23.json\nwriting: debug/nodes/1/ranges/24.json\nwriting: debug/nodes/1/ranges/25.json\nwriting: debug/nodes/1/ranges/26.json\nwriting: debug/nodes/1/ranges/27.json\nwriting: debug/nodes/1/ranges/28.json\nwriting: debug/nodes/1/ranges/29.json\nwriting: debug/nodes/1/ranges/30.json\nwriting: debug/nodes/1/ranges/31.json\nwriting: debug/nodes/1/ranges/32.json\nrequesting list of SQL databases... 3 found\nrequesting database details for defaultdb... writing: debug/schema/defaultdb@details.json.err.txt\n  ^- resulted in operation \"requesting database details for defaultdb\" timed out after 10s: rpc error: code = DeadlineExceeded desc = context deadline exceeded\nrequesting database details for postgres... writing: debug/schema/postgres@details.json\n0 tables found\nrequesting database details for system... writing: debug/schema/system@details.json\n26 tables found\nrequesting table details for system.comments... writing: debug/schema/system/comments.json\nrequesting table details for system.descriptor... writing: debug/schema/system/descriptor.json\nrequesting table details for system.eventlog... writing: debug/schema/system/eventlog.json\nrequesting table details for system.jobs... writing: debug/schema/system/jobs.json\nrequesting table details for system.lease... writing: debug/schema/system/lease.json\nrequesting table details for system.locations... writing: debug/schema/system/locations.json\nrequesting table details for system.namespace... writing: debug/schema/system/namespace.json\nrequesting table details for system.namespace2... writing: debug/schema/system/namespace2.json\nrequesting table details for system.protected_ts_meta... writing: debug/schema/system/protected_ts_meta.json\nrequesting table details for system.protected_ts_records... writing: debug/schema/system/protected_ts_records.json\nrequesting table details for system.rangelog... writing: debug/schema/system/rangelog.json\nrequesting table details for system.replication_constraint_stats... writing: debug/schema/system/replication_constraint_stats.json\nrequesting table details for system.replication_critical_localities... writing: debug/schema/system/replication_critical_localities.json\nrequesting table details for system.replication_stats... writing: debug/schema/system/replication_stats.json\nrequesting table details for system.reports_meta... writing: debug/schema/system/reports_meta.json\nrequesting table details for system.role_members... writing: debug/schema/system/role_members.json\nrequesting table details for system.role_options... writing: debug/schema/system/role_options.json\nrequesting table details for system.settings... writing: debug/schema/system/settings.json\nrequesting table details for system.statement_bundle_chunks... writing: debug/schema/system/statement_bundle_chunks.json\nrequesting table details for system.statement_diagnostics... writing: debug/schema/system/statement_diagnostics.json\nrequesting table details for system.statement_diagnostics_requests... writing: debug/schema/system/statement_diagnostics_requests.json\nrequesting table details for system.table_statistics... writing: debug/schema/system/table_statistics.json\nrequesting table details for system.ui... writing: debug/schema/system/ui.json\nrequesting table details for system.users... writing: debug/schema/system/users.json\nrequesting table details for system.web_sessions... writing: debug/schema/system/web_sessions.json\nrequesting table details for system.zones... writing: debug/schema/system/zones.json\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -47,3 +47,7 @@
        	            	 requesting heap profile for node 1... writing: debug/nodes/1/heap.pprof
        	            	-requesting heap files for node 1... 0 found
        	            	+requesting heap files for node 1... 4 found
        	            	+writing: debug/nodes/1/heapprof/memprof.000000000039162280_2020-04-22T02_36_38.141.pprof
        	            	+writing: debug/nodes/1/heapprof/memprof.000000000055517712_2020-04-22T02_36_48.142.pprof
        	            	+writing: debug/nodes/1/heapprof/memprof.000000000064787432_2020-04-22T02_36_58.142.pprof
        	            	+writing: debug/nodes/1/heapprof/memprof.000000000065956448_2020-04-22T02_37_18.142.pprof
        	            	 requesting goroutine files for node 1... 0 found
        	            	@@ -84,4 +88,4 @@
        	            	 requesting list of SQL databases... 3 found
        	            	-requesting database details for defaultdb... writing: debug/schema/defaultdb@details.json
        	            	-0 tables found
        	            	+requesting database details for defaultdb... writing: debug/schema/defaultdb@details.json.err.txt
        	            	+  ^- resulted in operation "requesting database details for defaultdb" timed out after 10s: rpc error: code = DeadlineExceeded desc = context deadline exceeded
        	            	 requesting database details for postgres... writing: debug/schema/postgres@details.json
        	Test:       	TestZip

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestZip PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestZip.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/cli/testdata/zip/testzip
+++ b/pkg/cli/testdata/zip/testzip
@@ -49,7 +49,7 @@ requesting data for debug/nodes/1/enginestats... writing: debug/nodes/1/enginest
 requesting stacks for node 1... writing: debug/nodes/1/stacks.txt
 requesting threads for node 1... writing: debug/nodes/1/threads.txt
 requesting heap profile for node 1... writing: debug/nodes/1/heap.pprof
-requesting heap files for node 1... 0 found
+requesting heap files for node 1... ? found
 requesting goroutine files for node 1... 0 found
 requesting log file ...
 requesting ranges... 32 found

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -128,6 +128,8 @@ func TestZip(t *testing.T) {
 	// Strip any non-deterministic messages.
 	out = eraseNonDeterministicZipOutput(out)
 
+	// We use datadriven simply to read the golden output file; we don't actually
+	// run any commands. Using datadriven allows TESTFLAGS=-rewrite.
 	datadriven.RunTest(t, "testdata/zip/testzip", func(t *testing.T, td *datadriven.TestData) string {
 		return out
 	})
@@ -264,6 +266,12 @@ func eraseNonDeterministicZipOutput(out string) string {
 	out = re.ReplaceAllString(out, `RPC connection to ...`)
 	re = regexp.MustCompile(`(?m)\^- resulted in.*$`)
 	out = re.ReplaceAllString(out, `^- resulted in ...`)
+
+	// The number of memory profiles previously collected is not deterministic.
+	re = regexp.MustCompile(`(?m)requesting heap files for node 1\.\.\..*found$`)
+	out = re.ReplaceAllString(out, `requesting heap files for node 1... ? found`)
+	re = regexp.MustCompile(`(?m)\^writing.*memprof*$`)
+	out = re.ReplaceAllString(out, ``)
 	return out
 }
 

